README.rst

```diff
@@ -190,6 +190,28 @@ for convenience use *payload* argument to mock out json response. Example below.
         # will throw an exception.
 
 
+**aioresponses allows to use callbacks to provide dynamic responses**
+
+.. code:: python
+
+    import asyncio
+    import aiohttp
+    from aioresponses import aioresponses
+
+    async def callback(match, url, *args, **kwargs):
+        return match.build_response(url, status=418)
```

Comment:
I think `ResponseData` is not the best name ;)
Name propositions:
`Response`
`ResponseData`
`AioResponse`
`Callback`
`Result` <- this one seems to be ok

any other propositions? :)

Reply:
```python
# example callback methods

def callback_tuple(url, **kw):
    return (201, {}, {"foo": "bar"})

def callback_result(url, **kw):
    return CallbackResult(status=201, body={"foo": "bar"})
```

Reply:
I think tuple is not the best solution as you may want to customize your response more heavily. Due to this I just moved `build_response` function from the `RequestMatch` class to the `compat` module and exposed it to users so inside this function can be any internal stuff.

What do you think about it?

Reply:
Hmm

Still I prefer approach "tell me what data do you want see in response"
rather than 
"If you decide to use callback you have to create response object manually"

so option with `CallbackResult` is more close to me ;)

with tuple I agree :)

Reply:
Addition to that I do not want to expose `compat` module to users  ;)
So I think `build_response` should remain "hidden" ;)